packages/icon/package.py

```diff
@@ -61,7 +65,8 @@ class Icon(Package):
     depends_on('netcdf-fortran')
     depends_on('netcdf-c')
     depends_on('cuda', when='icon_target=gpu')
-    depends_on('cdo', type='test')
+    depends_on('cdo')
```

Comment:
Why was it necessary to change cdo from a test dep to a build+link dep? (Deps are by default build+link)

Reply:
If we execute the test.py script outside of spack, concretization does not include CDO in its spec.
But cdo is needed for all tests in ICON.

Reply:
It also does not work when having cdo as test dep and calling the script from package.py of ICON.